controllers/install/common_helpers.go

```diff
@@ -54,3 +60,34 @@ func AllLabels(name string, labels map[string]string) map[string]string {
 	baseLabels = MergeMaps(baseLabels, identityLabels)
 	return baseLabels
 }
+
+// waitForJob will wait for some resolution of the job. Provide context with timeout if needed.
+func waitForJob(ctx context.Context, cl client.Client, job *batchv1.Job) (err error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.New("context timeout while waiting for job")
+		default:
+			key := client.ObjectKeyFromObject(job)
+			err = cl.Get(ctx, key, job)
+			if err != nil {
+				return err
+			}
+			if isJobFinished(job) {
+				return nil
+			}
+		}
+		time.Sleep(time.Second * 5)
+	}
+}
+
+// isJobFinished will assess if the job is finished (complete of failed).
+func isJobFinished(job *batchv1.Job) bool {
```

Comment:
I think we could use https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L85 or something like that.

Reply:
this is using the Status.Conditions, yes?

Reply:
added a test